File: java/spring-boot-example/src/main/java/com/creolecentric/example/exception/CreoleCentricException.java
package com.creolecentric.example.exception;

import lombok.Getter;

@Getter
public class CreoleCentricException extends RuntimeException {
    private final int statusCode;
    private final String responseBody;

    public CreoleCentricException(String message, int statusCode, String responseBody) {
        super(message);
        this.statusCode = statusCode;
        this.responseBody = responseBody;
    }

    public CreoleCentricException(String message) {
        this(message, 500, null);
    }
}

File: csharp/aspnet-core-example/Exceptions/CreoleCentricException.cs
namespace CreoleCentricExample.Exceptions;

public class CreoleCentricException : Exception
{
    public int StatusCode { get; }
    public string? ResponseBody { get; }

    public CreoleCentricException(string message, int statusCode = 500, string? responseBody = null)
        : base(message)
    {
        StatusCode = statusCode;
        ResponseBody = responseBody;
    }
}
